src/commands/moderation/prefixCommands/functions/deleteVersion.ts

```diff
@@ -0,0 +1,149 @@
+    color: Colors.Red,
+});
+
+const channelDefaultVersionPresentEmbed = makeEmbed({
+    title: 'Prefix Commands - Delete Version - Default Channel Versions Present',
+    description: 'There is one or more channel with this version selected as its default version. Please change or unset the default version for those channels first, or use the `force` option to delete the command version and all the default channel versions referencing it (making them default back to the GENERIC version).',
+    color: Colors.Red,
+});
+
+const failedEmbed = (versionId: string) => makeEmbed({
+    title: 'Prefix Commands - Delete Version - Failed',
+    description: `Failed to delete the prefix command version with id ${versionId}.`,
+    color: Colors.Red,
+});
+
+const doesNotExistsEmbed = (version: string) => makeEmbed({
+    title: 'Prefix Commands - Delete Version - Does not exist',
+    description: `The prefix command version ${version} does not exists. Can not delete it.`,
+    color: Colors.Red,
+});
+
+const successEmbed = (version: string, versionId: string) => makeEmbed({
+    title: `Prefix command version ${version} (${versionId}) was deleted successfully.`,
+    color: Colors.Green,
+});
+
+const modLogEmbed = (moderator: User, version: string, emoji: string, alias: string, enabled: boolean, versionId: string) => makeEmbed({
+    title: 'Prefix command version deleted',
+    fields: [
+        {
+            name: 'Version',
+            value: version,
+        },
+        {
+            name: 'Moderator',
+            value: `${moderator}`,
+        },
+        {
+            name: 'Emoji',
+            value: emoji,
+        },
+        {
+            name: 'Alias',
+            value: alias,
+        },
+        {
+            name: 'Enabled',
+            value: enabled ? 'Yes' : 'No',
+        },
+    ],
+    footer: { text: `Version ID: ${versionId}` },
+    color: Colors.Red,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Delete Version - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleDeletePrefixCommandVersion(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const version = interaction.options.getString('version')!;
+    const force = interaction.options.getBoolean('force') || false;
+    const moderator = interaction.user;
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
+        await interaction.followUp({ embeds: [noModLogs], ephemeral: true });
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.